routing/endpointregistry.go

```diff
@@ -115,23 +159,72 @@ func (r *EndpointRegistry) Do(routes []*Route) []*Route {
 	return routes
 }
 
+func (r *EndpointRegistry) updateStats() {
+	if !r.passiveHealthCheckEnabled {
+		return
+	}
+	ticker := time.NewTicker(r.statsResetPeriod)
+
+	for {
+		r.data.Range(func(key, value any) bool {
+			e := value.(*entry)
+
+			curSlot := e.curSlot.Load()
+			nextSlot := (curSlot + 1) % 2
+			e.totalFailedRoundTrips[nextSlot].Store(0)
+			e.totalRequests[nextSlot].Store(0)
+
+			failed := e.totalFailedRoundTrips[curSlot].Load()
+			requests := e.totalRequests[curSlot].Load()
+			failedRoundTripsRatio := float64(failed) / float64(requests)
```

Comment:
Can this become > 1?

Reply:
Lets acknowledge that here we have a race reading two values and add a check failedRoundTripsRatio = min(failedRoundTripsRatio, 1.0)